lang/java/reef-checkpoint/src/main/java/org/apache/reef/io/checkpoint/fs/FSCheckpointService.java

```diff
@@ -75,78 +75,90 @@ static final Path tmpfile(final Path p) {
   public CheckpointWriteChannel create()
       throws IOException {
 
-    final String name = namingPolicy.getNewName();
-
+    final String name = namingPolicy.getNewCheckpointName();
     final Path p = new Path(name);
     if (p.isUriPathAbsolute()) {
-      throw new IOException("Checkpoint cannot be an absolute path");
+      throw new IOException("Checkpoint name cannot be an absolute path.");
     }
-    return createInternal(new Path(base, p));
+
+    return createInternal(new Path(basePath, p));
   }
 
   CheckpointWriteChannel createInternal(final Path name) throws IOException {
-
     //create a temp file, fail if file exists
     return new FSCheckpointWriteChannel(name, fs.create(tmpfile(name), replication));
   }
 
   @Override
   public CheckpointReadChannel open(final CheckpointID id)
       throws IOException, InterruptedException {
+
     if (!(id instanceof FSCheckpointID)) {
       throw new IllegalArgumentException(
-          "Mismatched checkpoint type: " + id.getClass());
+          "Mismatched checkpoint id type. Expected FSCheckpointID, but actually got " + id.getClass());
     }
+
     return new FSCheckpointReadChannel(
         fs.open(((FSCheckpointID) id).getPath()));
   }
 
   @Override
-  public CheckpointID commit(final CheckpointWriteChannel ch) throws IOException,
-      InterruptedException {
+  public CheckpointID commit(final CheckpointWriteChannel ch)
+          throws IOException, InterruptedException {
+
     if (ch.isOpen()) {
       ch.close();
     }
+
     final FSCheckpointWriteChannel hch = (FSCheckpointWriteChannel) ch;
     final Path dst = hch.getDestination();
+
     if (!fs.rename(tmpfile(dst), dst)) {
       // attempt to clean up
       abort(ch);
       throw new IOException("Failed to promote checkpoint" +
           tmpfile(dst) + " -> " + dst);
     }
+
     return new FSCheckpointID(hch.getDestination());
   }
 
   @Override
   public void abort(final CheckpointWriteChannel ch) throws IOException {
+
     if (ch.isOpen()) {
       ch.close();
     }
+
     final FSCheckpointWriteChannel hch = (FSCheckpointWriteChannel) ch;
     final Path tmp = tmpfile(hch.getDestination());
+
     try {
       if (!fs.delete(tmp, false)) {
-        throw new IOException("Failed to delete checkpoint during abort");
+        throw new IOException("Failed to delete a temporary checkpoint file during abort. Path: " + tmp);
       }
     } catch (final FileNotFoundException ignored) {
       // IGNORE
```

Comment:
Why can we igore the exception here and in the similar case in the delete() method ?